AsyncDisplayKit/Details/ASCollectionViewFlowLayoutInspector.m

```diff
@@ -0,0 +1,121 @@
+/* Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <UIKit/UIKit.h>
+
+#import "ASCollectionViewFlowLayoutInspector.h"
+
+#import "ASCollectionView.h"
+
+@implementation ASCollectionViewFlowLayoutInspector {
+  BOOL _delegateImplementsReferenceSizeForHeader;
+  BOOL _delegateImplementsReferenceSizeForFooter;
+}
+
+#pragma mark - Accessors
+
+- (instancetype)initWithCollectionView:(ASCollectionView *)collectionView flowLayout:(UICollectionViewFlowLayout *)flowLayout;
+{
+  self = [super init];
+
+  if (flowLayout == nil) {
```

Comment:
Probably ASDisplayNodeAssert(NO, @"Should never create a layout inspector without a layout") ?  Great of course to have the return nil for safer production runtime behavior.